stix_shifter_modules/nozomi/stix_transmission/connector.py

```diff
@@ -0,0 +1,216 @@
+                        # Handling invalid query exception
+                        if 'Query is not valid' in response_dict.get('error', ''):
+                            return_obj = self.exception_response(response_wrapper.code, response_dict.get('error'))
+                            data = []
+                            break
+
+                        return_obj['success'] = True
+                        data += response_dict['result']
+                        result_count += len(response_dict['result'])
+                        local_result_count += len(response_dict['result'])
+
+                        if response_dict['total'] <= result_count or len(response_dict['result']) == 0:
+                            next_url = None
+                            break
+                        # parsing the next_url
+                        page = regex.findall(r'page=([^>]*?)&', query_url)[0]
+                        next_url = regex.sub(r'page=([^>]*?)&', f'page={str(int(page) + 1)}&', query_url)
+
+                        if not metadata and result_count < total_records:
+                            remaining_records = total_records - result_count
+                        elif metadata and local_result_count < total_records:
+                            remaining_records = total_records - local_result_count
+                        else:
+                            break
+
+                        if remaining_records > Connector.NOZOMI_MAX_PAGE_SIZE:
+                            page_size = Connector.NOZOMI_MAX_PAGE_SIZE
+                        else:
+                            page_size = remaining_records
+                    # Handling expired token
+                    elif response_wrapper.code == 401 and not token_generated and \
+                            (response == '' or
+                             'Signature has expired' in response_dict.get('error', {}).get('message', '')):
+                        jwt_token = await self.__get_token()
+                        token_generated = True
+                        continue
+                    else:
+                        error = response_dict.get('error', '')
+                        if response_wrapper.code == 403:
+                            error = 'Query length is too long or Invalid Query'
+                        return_obj = self.exception_response(response_wrapper.code, error)
+                        data = []
+                        break
+
+                if data:
+                    data = self.get_results_data(data)
+                    if metadata:
+                        return_obj['data'] = data if data else []
+                    else:
+                        return_obj['data'] = data[int(offset): total_records] if data else []
+                    if jwt_token and next_url and result_count < self.api_client.result_limit:
+                        return_obj['metadata'] = {"result_count": result_count, "jwtToken": jwt_token,
```

Comment:
We can't return the jwt token in the return_obj. Doing so could be a security risk as the jwt token is still credentials, even if they are short lived. Putting the jwt token into the meta data exposes it in a way we do not want to do.

You'll need to make some modifications in your code to account for this change.

Reply:
Most of our previous connectors kept tokens in metadata. If we don't want to do this, we need to generate a JWT token for each request. Please confirm if this is fine, and we can proceed with this development.

Reply:
Yes, it's fine to create a new JWT token on each request for data. Just to be clear, you can reuse the token within the "create_results_connection" method, you just can't post it in the return_obj.

Reply:
Removed jwt token in metadata. So here after every API call will generate new jwt token.